packages/vscale/src/utils/tick-sample.ts

```diff
@@ -450,3 +471,65 @@ export function parseNiceOptions(originalDomain: number[], option: NiceOptions)
 
   return { niceType, niceDomain, niceMinMax, domainValidator };
 }
+
+export const ticksBaseTransform = memoize<TicksBaseTransformFunc>(
+  (
+    start: number,
+    stop: number,
+    count: number,
+    base: number,
+    transformer: TransformType,
+    untransformer: TransformType
+  ) => {
+    const ticksResult: number[] = [];
+    const ticksMap = {};
+    const startExp = transformer(start);
+    const stopExp = transformer(stop);
+    let ticksExp = [];
+    // get ticks exp
+    if (Number.isInteger(base)) {
+      ticksExp = new LinearScale().domain([startExp, stopExp]).ticks(count);
```

Comment:
dont create `LinearScale` here, just call `ticks()` function

this line will produce circular dependencies